### PR TITLE
fix:import testing framework for few tools scripts

### PR DESCRIPTION
**Pull request by sudanl0**
## Changes

Make testing framework available so that the few tools scripts using them work as expected.

## Reason

changes added in #4089 breaks cross snapshot restore tests.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] ~If a specific issue led to this PR, this PR closes the issue.~
- [x] The description of changes is clear and encompassing.
- [ ] ~Any required documentation changes (code and docs) are included in this PR.~
- [ ] ~API changes follow the [Runbook for Firecracker API changes][2].~
- [ ] ~User-facing changes are mentioned in `CHANGELOG.md`.~
- [x] All added/changed functionality is tested.
- [ ] ~New `TODO`s link to an issue.~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by roypat — DISMISSED** [review]
Ooops, thanks for fixing!

File: tools/create_snapshot_artifact/main.py [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
-from host_tools.cargo_build import get_firecracker_binaries
-
 # Hack to be able to import testing framework functions.
 sys.path.append(os.path.join(os.getcwd(), "tests"))  # noqa: E402
 
 # pylint: disable=wrong-import-position
+from host_tools.cargo_build import get_firecracker_binaries
+
 # The test infra assumes it is running from the `tests` directory.
 os.chdir("tests")
```
Comment (roypat) [inline tools/create_snapshot_artifact/main.py:21]: Actually, it seems there's already a workaround for this (didnt know that when I wrote my ab_test.py script). Can we use the same hack everywhere? E.g. either the "os.chdir" to get it to resolve the imports, or the `path.append`?

File: tools/test-popular-containers/test-docker-rootfs.py [hunk 2]
```diff
@@ -8,12 +8,20 @@
 Test all the ext4 rootfs in the current directory
 """
 
+import os
+import sys
 from pathlib import Path
 
+# Hack to be able to import testing framework functions.
+sys.path.append(os.path.join(os.getcwd(), "tests"))  # noqa: E402
```
Comment (pb8o) [inline tools/test-popular-containers/test-docker-rootfs.py:16]: I think this one is working fine?

File: tools/create_snapshot_artifact/main.py (hunk 1 above)
Comment (pb8o) [reply tools/create_snapshot_artifact/main.py:21]: There's another way, we could add `env PYTHONPATH=tests` here: https://github.com/firecracker-microvm/firecracker/blob/main/.buildkite/pipeline_cross.py#L42

Or we could even add it always `devtool::cmd_sh`

Comment (sudanl0) [reply tools/create_snapshot_artifact/main.py:21]: The `env PYTHONPATH=tests` thing was the first solution I tried but then I saw that there was an existing fix with the sys.path.append and with this we can run the scripts standalone inside the container as well.

**Review by roypat — APPROVED** [review]

File: tools/test-popular-containers/test-docker-rootfs.py (hunk 2 above)
Comment (sudanl0) [reply tools/test-popular-containers/test-docker-rootfs.py]: do you mean if we need the fix in this script? if so yes, the script doesn't work without the fix.

Comment (sudanl0) [reply tools/test-popular-containers/test-docker-rootfs.py]: yes pipeline is working, for ref. I checked it here:  https://buildkite.com/firecracker/sudan-tests/builds/10#018acb83-eee4-4b61-90be-1b97b81668a8

File: tools/create_snapshot_artifact/main.py (hunk 1 above)
Comment (pb8o) [reply tools/create_snapshot_artifact/main.py]: I am more partial to the environment variable, but this usually has worked well in the past, so no reason to change it.

**Review by pb8o — APPROVED** [review]

**Pull request merged**